applications/asset_tracker_v2/tests/json_common/src/main.c

```diff
@@ -0,0 +1,1175 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <ztest.h>
+#include <zephyr.h>
+#include <stdio.h>
+#include <string.h>
+#include <cJSON.h>
+#include <cJSON_os.h>
+
+#include "json_helpers.h"
+#include "json_common.h"
+#include "cloud_codec.h"
+#include "json_protocol_names.h"
+#include "json_validate.h"
+
+/* Structure used to generate cJSON objects and encoded output string buffers. */
+struct test_dummy {
+	cJSON *root_obj;
+	cJSON *array_obj;
+	char *buffer;
+} dummy;
+
+/* Helper functions */
+
+/* Function used to find the first unmatched character when comparing two strings.
+ * Modified strcmp.
+ */
+int find_index_first_unmatched_character(const char *string1, const char *string2)
+{
+	int index = 0;
+
+	while ((*string1 == *string2) && (*string1 != '\0')) {
+		string1++;
+		string2++;
+
+		index++;
+	}
+
+	return index;
+}
+
+/* Function used to check the return value from the encoding functions in JSON common API and the
+ * encoded output.
+ */
+static void encoded_output_check(int ret, cJSON *object, char *validation_string,
+				 const char *func_name, int8_t queued)
+{
+	zassert_equal(0, ret, "Function %s does not return 0, returns %d", func_name, ret);
```

Comment:
I think I prefer asserting where the error originates, makes things a bit simpler to follow

Reply:
It would lead to a lot more duplicate code (250 lines extra ish). But, I think readability should outweigh lines of code. So I don't mind doing the change. Currently, both ztest and ```encoded_output_check()``` will print the name of the function that fails which should be enough. But of course, you don't immediately see where the tests fails due to the asserted line number being somewhat useless in this case.

Reply:
I was thinking about only this line, so that `ret` doesn't need to be passed in here and checked in this function. It could be checked where it's assigned. I don't think that should add many lines

Reply:
Ah, I see. I misunderstood.